src/it/unimi/dsi/fastutil/io/FastMultiByteArrayInputStream.java

```diff
@@ -76,14 +76,28 @@ public FastMultiByteArrayInputStream(final InputStream is, long size) throws IOE
 		for(int i = 0; i < array.length - 1; i++) {
 			array[i] = new byte[size >= SLICE_SIZE ? SLICE_SIZE : (int)size];
 			// It is important *not* to use is.read() directly because of bug #6478546
-			if (BinIO.loadBytes(is, array[i]) != array[i].length) throw new EOFException();
+			if (read(is, array[i]) != array[i].length) throw new EOFException();
```

Comment:
I'm not sure this can be done. Bug #6478546 is still around in old JVMs. We might consider copying the code, but we cannot use just read().